File: backend/models/learning_content.py
"""Pydantic models for LearningContent."""

from pydantic import BaseModel, Field

class LearningContentBase(BaseModel):
    title: str = Field(..., example="과학 탐험")
    content: str = Field(..., example="과학 탐험 내용입니다.")
    difficulty: str = Field(..., example="중급")

class LearningContentCreate(LearningContentBase):
    pass

class LearningContent(LearningContentBase):
    id: int

    class Config:
        orm_mode = True

File: lib/models/learning_content.dart
class LearningContent {
  final String title;
  final String description;
  final int difficulty;

  const LearningContent({required this.title, required this.description, required this.difficulty});

  Map<String, dynamic> toJson() => {
        'title': title,
        'description': description,
        'difficulty': difficulty,
      };
}
